src/test/java/uk/gov/hmcts/ccd/endpoint/ui/QueryEndpointIT.java

```diff
@@ -114,7 +114,7 @@ public void validSearch() throws Exception {
         // Check that we have the expected test data set size, this is to ensure
```

Comment:
CaseDetailsQueryBuilder is rather complex, I would add unit tests for it. We can't rely only on Integration Tests

Reply:
I would as well, but I cannot think of a decent way of doing it because of the Criteria API. One way would be to create a rather complex hierarchy of mocks; the other way would be to use a real persistence context, build query, unwrap them into the JPA provider specific type and then output them as SQL and test the string. I'm not sold on either of those :s